src/libfetchers/memory-input-accessor.hh

```diff
@@ -0,0 +1,15 @@
+#include "input-accessor.hh"
+
+namespace nix {
+
+/**
+ * An input accessor for an in-memory file system.
+ */
+struct MemoryInputAccessor : InputAccessor
+{
+    virtual SourcePath addFile(CanonPath path, std::string && contents) = 0;
+};
+
+ref<MemoryInputAccessor> makeMemoryInputAccessor();
+
+}
```

Comment:
I think we soon want want

1. Move this to `libutil`, make it a `SourceAccessor`
2. Use it in `DummyStore`, so that that thing can implement far more methods, rename it t o`MemoryStore`
3. For dummy store without in-memory use, we can just use read-only mode.

I am not insisting we should do all of that in this PR, but it would help us unit-test a lot of things which are currently untested.

Reply:
Yeah that sounds good (in a future PR).